### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by utkarshayachit**
fixes various warnings popping up on vtk-m dashboards.

### CODE REVIEW

File: include/diy/master.hpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ namespace diy
       using Skip = std::function<bool(int, const Master&)>;
 
       struct SkipNoIncoming;
-      struct NeverSkip { bool    operator()(int i, const Master& master) const   { return false; } };
+      struct NeverSkip { bool    operator()(int i, const Master& master) const   { (void)i; (void)master; return false; } };
```
Comment (mrzv) [inline include/diy/master.hpp]: Let's be consistent and simply not name these kinds of unused arguments in the function's parameter list.

File: include/diy/master.hpp [hunk 2]
```diff
@@ -80,8 +80,8 @@ namespace diy
       struct QueueSizePolicy: public QueuePolicy
       {
                 QueueSizePolicy(size_t sz): size(sz)          {}
-        bool    unload_incoming(const Master& master, int from, int to, size_t sz) const    { return sz > size; }
-        bool    unload_outgoing(const Master& master, int from, size_t sz) const            { return sz > size*master.outgoing_count(from); }
+        bool    unload_incoming(const Master& master, int from, int to, size_t sz) const    { (void)master; (void)from; (void)to; return sz > size; }
```
Comment (mrzv) [inline include/diy/master.hpp]: Same here.

File: include/diy/master.hpp [hunk 3]
```diff
@@ -80,8 +80,8 @@ namespace diy
...
+        bool    unload_outgoing(const Master& master, int from, size_t sz) const            { (void)master; (void)from; return sz > size*master.outgoing_count(from); }
```
Comment (mrzv) [inline include/diy/master.hpp]: And here.

File: include/diy/master.hpp [hunk 4]
```diff
@@ -703,6 +703,7 @@ diy::Master::
 foreach_(const Callback<Block>& f, const Skip& skip)
 {
     auto scoped = prof.scoped("foreach");
+    (void)scoped;
```
Comment (mrzv) [inline include/diy/master.hpp]: For these kinds of unused variables, let's add a macro `DIY_UNUSED` and label them with it. Here's a version of it from `examples/simple/simple.cpp`:
```
#define UNUSED(expr) do { (void)(expr); } while (0)
```

**Comment by utkarshayachit** [comment]
will do. I'll update this once the MPI branch is merged as I am sure the two will conflict.

**Comment by utkarshayachit** [comment]
@mrzv this has been updated as you suggested. Let me know what you think. Thanks.

**Comment by mrzv** [comment]
Looks great. I'm merging it.

**Comment by utkarshayachit** [comment]
sweet! thanks!

**Pull request merged**